RidesHandler.py

```diff
@@ -1,101 +1,186 @@
+from datetime import datetime
+
+import arrow
+from requests import RequestException
+
 import DBHandler
 import UsersHandler
 import log
 from consts import *
-from requests import RequestException
-from datetime import datetime
-# import nexmo
-#
-# client = nexmo.Client(key='82a7fce6', secret='3dc02eec783b28cc')
-#
-# client.send_message({
-#     'from': 'IDF Carpool',
-#     'to': 'number',
-#     'text': 'Hi!\n You have a ride request\n :)',
-# })
 
 logger = log.setup_custom_logger()
 firebase_db = DBHandler.get_firebase_db_ref()
 
 
 def get_all_rides_from_db():
     logger.info('Trying to get all rides from DB')
-    return firebase_db.child(rides_collection).get().items()
+    all_rides = firebase_db.child(rides_collection).get()
 
+    if all_rides is not None:
+        return all_rides.items()
 
-def find_a_ride(ride_request):
-    optional_drivers = get_all_optional_drivers(ride_request)
+    return None
 
-    if optional_drivers:
-        logger.info("Founded %s optional drivers for the ride", len(optional_drivers))
-        logger.info("Returning the optional drivers")
 
-        return optional_drivers
-    else:
-        logger.info("No optional drivers for this ride")
+def parse_time_to_hour_minutes_seconds(ride_time):
+    parsed_ride_hour = arrow.get(ride_time).to(time_zone).datetime.hour
+    parsed_ride_minutes = arrow.get(ride_time).to(time_zone).datetime.minute
+
+    return str(parsed_ride_hour) + ':' + str(parsed_ride_minutes)
+
+
+def calc_time_difference_in_minutes(driver_time, ride_time):
+    parsed_ride_time = parse_time_to_hour_minutes_seconds(ride_time)
 
-        return None
+    time_format = '%H:%M'
+    time_diff = datetime.strptime(parsed_ride_time, time_format) - datetime.strptime(driver_time, time_format)
+    if time_diff.days < 0:
+        time_diff = datetime.strptime(driver_time, time_format) - datetime.strptime(parsed_ride_time, time_format)
+
+    return time_diff.seconds / 60
+
+
+def is_matching_hours(driver_time, ride_time):
+    return calc_time_difference_in_minutes(driver_time, ride_time) <= 60
+
+
+def is_driver_is_optional(driver, ride_request):
+    if ride_request.get(home_to_base):
+        return driver.get(home_location) == ride_request.get(source) and \
+               driver.get(base_location) == ride_request.get(destination) and \
+               is_matching_hours(driver.get(leaving_home_time), ride_request.get(date))
+    else:
+        return driver.get(base_location) == ride_request.get(source) and \
+               driver.get(home_location) == ride_request.get(destination) and \
+               is_matching_hours(driver.get(leaving_base_time), ride_request.get(date))
```

Comment:
The only thing different in if's is driver.get(leaving_base_time) instead of driver.get(leaving_home_time)
Extract that to a variable and do a single return